zb-db/src/main/java/io/camunda/zeebe/db/impl/rocksdb/transaction/TransactionalColumnFamily.java

```diff
@@ -332,11 +352,7 @@ private void forEachInPrefix(
 
             boolean shouldVisitNext = true;
 
-            for (RocksDbInternal.seek(
-                    iterator,
-                    ZeebeTransactionDb.getNativeHandle(iterator),
-                    prefixKey,
-                    prefixLength);
+            for (iterator.seek(columnFamilyContext.keyWithColumnFamily(startAt));
```

Comment:
@oleschoenburg, just out of curiosity, did you also try to set a `iterateLowerBound` instead of directly seeking the start key. To my understanding, RocksDB will use the lower bound for the seek, if the lower bound is greater then the seek target. Basically, it will resolve it internally. To set a lower bound would be like this

```
transactionDb.getPrefixReadOptions().setIteratorLowerBound(...)
```
Probably it doesn't make any difference, just wanted to check.

Reply:
Hmm but the docs say: "Defines the smallest key at which the backward iterator can return an entry". It does not specify that this is used for the initial seek.

Reply:
We also don't do any backward iteration :thinking:

Reply:
Yeah, this doesn't work without the seek.

Reply:
Sorry, I wasn't that clear on this. I know, that Zeebe does not do backward iteration, and yes, it still requires the seek. The code could look like this:

```
try (final RocksIterator iterator = newIterator(context, transactionDb.getPrefixReadOptions().setIteratorLowerBound(...))) {
  boolean shouldVisitNext = true;
  for (RocksDbInternal.seek(
                    iterator,
                    ZeebeTransactionDb.getNativeHandle(iterator),
                    prefixKey,
                    prefixLength); {
    ...
  }
}
```

With that, RocksDB will set the iterator to the expected position during seek. It basically, checks if the lower bound is bigger than the passed target seek position, then it will use the lower bound as a target for the seek. If the lower bound is smaller, then it will keep the passed target for the seek.

I just think the way the PR proposes to implement it might be a problem in running the iterator in other modes like [adaptive mode](https://github.com/facebook/rocksdb/wiki/Prefix-Seek#adaptive-prefix-mode) (which we don't do of course).

However, if it works the way it is implemented now, I am okay with that, just wanted to check if you tried the other approach with lower bound as well.